core/unix/signal.c

```diff
@@ -3178,9 +3178,17 @@ convert_frame_to_nonrt(dcontext_t *dcontext, int sig, sigframe_rt_t *f_old,
     memcpy(&f_new->sc, get_sigcontext_from_rt_frame(f_old), sizeof(sigcontext_t));
     if (sc_old->fpstate != NULL) {
         /* up to caller to include enough space for fpstate at end */
+        uint offset_fpstate_xsave = offsetof(kernel_fpstate_t, _fxsr_env);
+        /* i#5079: The kernel requires the xsave area to be 64-byte aligned, which
+         * starts at _fxsr_env in kernel_fpstate_t.
```

Comment:
Please add a comment to _fxsr_env in sigcontext.h.  I see from the comments there that we added the prior fields to the struct ourselves (the kernel treated those separately), which makes it seem a little better here: it's only DR who has this strange internal field alignment.  Please add to those prepended-field comments too that adding them forces us to align the combined struct for an internal field here.

Reply:
> Please add to those prepended-field comments too that adding them forces us to align the combined struct for an internal field here

Do you mean at L97 in sigcontext.h? I noted the alignment requirement at L116. Let me know if more comments are needed.

Reply:
LGTM